### PR TITLE
Adding IMip scheduling which uses ownClouds mail delivery

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@georgehrke @raghunayyar please review and test - THX

### CODE REVIEW

File: apps/dav/lib/caldav/schedule/imipplugin.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+		$message->setReplyTo([$sender => $senderName])
+			->setTo([$recipient => $recipientName])
+			->setSubject($subject)
+			->setBody($iTipMessage->message->serialize(), $contentType);
+		try {
+			$failed = $this->mailer->send($message);
+			if ($failed) {
+				$this->logger->error('Unable to deliver message to {failed}', ['app' => 'dav', 'failed' =>  implode(', ', $failed)]);
+				$iTipMessage->scheduleStatus = '5.0; EMail delivery failed';
```
Comment (DeepDiver1975) [inline apps/dav/lib/caldav/schedule/imipplugin.php]: @evert is this the proper behavior in case of error? THX

Comment (evert) [inline apps/dav/lib/caldav/schedule/imipplugin.php]: Looks good to me!

Comment (DeepDiver1975) [inline apps/dav/lib/caldav/schedule/imipplugin.php]: > Looks good to me!

THX

**Comment by DeepDiver1975** [comment]
@LukasReschke @MorrisJobke @PVince81 @blizzz @schiesbn please review - THX

File: apps/dav/lib/caldav/schedule/imipplugin.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+		if (!$iTipMessage->significantChange) {
+			if (!$iTipMessage->scheduleStatus) {
+				$iTipMessage->scheduleStatus = '1.0;We got the message, but it\'s not significant enough to warrant an email';
+			}
+			return;
+		}
+
+		$summary = $iTipMessage->message->VEVENT->SUMMARY;
+
+		if (parse_url($iTipMessage->sender, PHP_URL_SCHEME) !== 'mailto')
```
Comment (georgehrke) [inline apps/dav/lib/caldav/schedule/imipplugin.php]: please use enclosing brackets

**Comment by georgehrke** [comment]
:+1: except for the one issue

**Comment by DeepDiver1975** [comment]
Test sceanrio:
1. connect mac calendar app to a server running this code base
2. make sure email settings work
3. create new appointment in calendar on mac
4. add an email address of somebody you want to invite
5. wait for the email to arrive - the email should contain an ics file with the appointment

**Comment by MorrisJobke** [comment]
Test sceanrio:
1. connect mac calendar app to a server running this code base :white_check_mark: 
2. make sure email settings work :white_check_mark: 
3. create new appointment in calendar on mac :white_check_mark: 
4. add an email address of somebody you want to invite :white_check_mark: 
5. wait for the email to arrive - the email should contain an ics file with the appointment :x:

**Comment by MorrisJobke** [comment]
Nothing in the logs. I tested two different email addresses

**Comment by DeepDiver1975** [comment]
> Nothing in the logs. I tested two different email addresses

please enable mail debug logging and watchout for log entries which show that the mails are sent - or not

**Comment by DeepDiver1975** [comment]
> please enable mail debug logging and watchout for log entries which show that the mails are sent - or not

@MorrisJobke mind taking a second look? THX

**Comment by MorrisJobke** [comment]
> @MorrisJobke mind taking a second look? THX

I'm on it.

**Comment by DeepDiver1975** [comment]
> I'm on it.

THX a lot!

**Comment by MorrisJobke** [comment]
Debugged this together with @DeepDiver1975 and he will have a look at the found issues.

**Comment by MorrisJobke** [comment]
I don't get any emails delivered.

That is the requests that is made (http://192.168.0.12/master/remote.php/dav/calendars/mjob/outbox/):
bildschirmfoto 2016-01-08 um 17 18 13


```ical
BEGIN:VCALENDAR
CALSCALE:GREGORIAN
VERSION:2.0
METHOD:REQUEST
PRODID:-//Apple Inc.//Mac OS X 10.11.2//EN
BEGIN:VFREEBUSY
UID:C8F8DA55-70FB-4A59-A034-484A559F63A8
DTEND:20160113T090000Z
X-CALENDARSERVER-EXTENDED-FREEBUSY:T
ATTENDEE:mailto:mail@example.com
DTSTART:20160113T080000Z
X-CALENDARSERVER-MASK-UID:B20D4FC7-51C3-4956-BC70-E1983B7D8702
DTSTAMP:20160108T161534Z
ORGANIZER:/master/remote.php/dav/principals/users/mjob/
END:VFREEBUSY
END:VCALENDAR
```

**Comment by MorrisJobke** [comment]
The response is:

```
<cal:schedule-response xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:cal="urn:ietf:params:xml:ns:caldav" xmlns:cs="http://calendarserver.org/ns/" xmlns:card="urn:ietf:params:xml:ns:carddav" xmlns:oc="http://owncloud.org/ns"> 
	<cal:response> 
		<cal:recipient> 
			<d:href>mailto:mail@example.com</d:href> 
		</cal:recipient> 
		<cal:request-status>3.7;Could not find principal</cal:request-status> 
	</cal:response> 
</cal:schedule-response>
```

(I replaced my email address with mail@example.com in the above snippets.

**Comment by MorrisJobke** [comment]
It now works. I used my own email address. Now I used a different one and it works.

:+1:

**Pull request merged**